chore: Prepare 1.230.0 release (#460)
newrelic/newrelic-browser-agent: CHANGELOG.md

Reviewer: `setCustomAttribute` update memo missing in 1.230

## v1.230.0

### Add persistence for custom attributes

> Added the ability for custom attributes to be persisted between page loads via session storage. The API `setCustomAttribute` takes a new, third parameter. The new parameter defaults to `false`. When passed a `true` boolean, the provided attribute will be persisted in session storage and restored on subsequent page loads. Session storage is subject to the [same-origin policy](https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy) of the browser.

Reviewer 2: ```suggestion
Custom attributes may now optionally be persisted between page loads via session storage. The API method `setCustomAttribute` now takes a third parameter, which defaults to `false`. When `true`, this boolean indicates that the provided custom attribute should be persisted in session storage and restored on subsequent page loads. Session storage is subject to the [same-origin policy](https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy) of the browser.
```

### Add setUserId API

> Added a new API method `setUserId` that defines a custom attribute containing the provided user ID. The user ID will be included in all payloads. The `setUserId` API is an extension of the `setCustomAttribute` API and automatically persists between page loads.

Reviewer 2: ```suggestion
A new `setUserId` API method is now available. It defines a custom attribute containing a specified user ID, to be included in all payloads. The `setUserId` API is an extension of the `setCustomAttribute` API and automatically persists between page loads.
```

### Fix npm package browser targets
The previous npm package release included code that was not compatible with browsers in our [supported browser list](https://docs.newrelic.com/docs/browser/new-relic-browser/getting-started/compatibility-requirements-browser-monitoring/#browser-types). The package exports will now default to code that matches our browser support list. The package will also include the source code for power users that would like more control over how the agent is built.

### Fix npm package exports
Removed some dangling, unused, and non-existent imports and exports from the source code that was causing issues with consuming the NPM package in some projects.

### Fix npm package import for typescript projects
Fixed an issue where typescript projects attempting to import the browser agent library would not get proper auto-complete in the import statement and would receive module declaration errors. The browser agent will not use tsc to generate type definitions from the JavaScript code at build time. The package.json has been updated with appropriate type export declarations.

### Add error catch around sendBeacon

> Added error catch logic around the sendBeacon usage. In the event sendBeacon throws an error during the final harvest, the final harvest will attempt to fallback to image data submission method.

Reviewer 2: ```suggestion
Added error catch logic around the sendBeacon usage. In the event sendBeacon throws an error during the final harvest, the final harvest will attempt to fall back to the image data submission method.
```

### Removed array reduce polyfill

> Removed the array reduce polyfill in favor of native browser array reduce method.

Reviewer 2: ```suggestion
Removed the array reduce polyfill in favor of the native browser array reduce method.
```
